src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1455,71 +1478,72 @@ class StatementSegment(BaseSegment):
 class JoinClauseSegment(BaseSegment):
     """Any number of join clauses, including the `JOIN` keyword.
 
-    https://spark.apache.org/docs/3.0.0/sql-ref-syntax-qry-select-join.html
-    TODO: Add NATURAL JOIN syntax.
+    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-join.html
     """
 
     type = "join_clause"
-    match_grammar = Sequence(
+
+    match_grammar = StartsWith(
+        OneOf(
+            Sequence(
+                Ref("JoinTypeKeywords", optional=True),
+                Ref("JoinKeywords"),
+            ),
+            Sequence(Ref("NaturalJoinKeywords"), Ref("JoinKeywords")),
+        ),
+        Ref("FromExpressionElementSegment"),
```

Comment:
I wonder if this could be simplified:

```suggestion
        OneOf(
            Ref("JoinTypeKeywords"),
            Ref("JoinKeywords"),
            Ref("NaturalJoinKeywords"),
        ),
```